dual_screen/android/src/main/kotlin/com/builttoroam/dual_screen/DualScreenPlugin.kt

```diff
@@ -0,0 +1,132 @@
+
+    fun registerPlugin(context: Context?, messenger: BinaryMessenger?) {
+        activity = this.activity
+        this.context = context
+        MethodChannel(messenger, METHOD_CHANNEL_NAME).setMethodCallHandler(this)
+        EventChannel(messenger, EVENT_CHANNEL_NAME).setStreamHandler(this)
+    }
+
+    override fun onAttachedToEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        this.registerPlugin(binding.getApplicationContext(), binding.getBinaryMessenger())
+    }
+
+    companion object {
+        @JvmStatic
+        fun registerWith(registrar: Registrar) {
+            DualScreenPlugin().registerPlugin(registrar.context(), registrar.messenger())
+        }
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "isDualScreenDevice" -> {
+                result.success(isDualScreenDevice())
+            }
+            "isAppSpanned" -> {
+                if (isDualScreenDevice()) {
+                    result.success(isAppSpanned())
+                } else {
+                    result.success(false)
+                }
+            }
+            else -> {
+                result.notImplemented()
+            }
+        }
+    }
+
+    override fun onListen(arguments: Any?, events: EventSink?) {
+        activity?.getWindow()?.getDecorView()?.getRootView()?.addOnLayoutChangeListener(object : OnLayoutChangeListener {
+            override fun onLayoutChange(
+                v: View?,
+                left: Int,
+                top: Int,
+                right: Int,
+                bottom: Int,
+                oldLeft: Int,
+                oldTop: Int,
+                oldRight: Int,
+                oldBottom: Int
+            ) {
+                events!!.success(isAppSpanned())
+            }
+        })
+    }
+
+    override fun onCancel(arguments: Any?) {
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        this.activity = binding.activity
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        this.activity = binding.activity
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        this.activity = null
+    }
+
+    override fun onDetachedFromActivity() {
+        this.activity = null
+    }
+
+    fun isDualScreenDevice(): Boolean {
+        return context?.getPackageManager()!!.hasSystemFeature(
+            "com.microsoft.device.display.displaymask"
+        )
+    }
+
+    fun isAppSpanned(): Boolean {
+        if (isDualScreenDevice()) {
+            var boundings = DisplayMask.fromResourcesRectApproximation(this.activity).getBoundingRects()
+            if (boundings.isEmpty()) {
+                return false
+            }
+            var drawingRect = android.graphics.Rect()
+            activity?.getWindow()?.getDecorView()?.getRootView()?.getDrawingRect(drawingRect)
```

Comment:
I'm puzzled by this activity vs this.activity again. Maybe I just don't understand how dart plugins work but do we need both and how do they differ?

Reply:
Using 'this' is just explicit that we want the activity defined in the class. Usually helpful when you have a variable named the same thing that you want to assign to. Originally these methods had an activity parameter but has been changed to binding.activity so ive removed 'this' references